src/main/scala/pt/tecnico/dsi/openstack/keystone/services/Domains.scala

```diff
@@ -28,18 +27,27 @@ final class Domains[F[_]: Sync: Client](baseUri: Uri, authToken: Header)
     )))
 
   /**
-    * Get detailed information about the domain specified by name.
+    * Get detailed information about the domain specified by name, assuming it exists.
     *
     * @param name the domain name
-    * @return the domain matching the name.
+    * @return the domain matching the name. If none exists F will contain an error.
     */
-  def getByName(name: String): F[Domain] = {
+  def applyByName(name: String): F[Domain] = {
     // A domain name is globally unique across all domains.
     list(Query.fromPairs("name" -> name)).compile.lastOrError
   }
+  /**
+   * Get detailed information about the domain specified by name.
+   *
+   * @param name the domain name
+   * @return a Some of the domain matching the name if it exists. A None otherwise.
+   */
+  def getByName(name: String): F[Option[Domain]] =
```

Comment:
Codacy found an issue: [Do not prefix getters with 'get'](https://app.codacy.com/gh/ist-dsi/scala-keystone-client/pullRequest?prid=6017598)